power_grid_model_c/power_grid_model/include/power_grid_model/job_adapter.hpp

```diff
@@ -83,7 +95,7 @@ class JobAdapter<MainModel, ComponentList<ComponentType...>>
     // current_scenario_sequence_cache_ is calculated per scenario, so it is excluded from the constructors.
     main_core::utils::SequenceIdx<ComponentType...> current_scenario_sequence_cache_{};
 
-    std::mutex calculation_info_mutex_;
+    Logger* log_{nullptr};
```

Comment:
Why does job adapter also need to store it?

Reply:
i've thought about his for a bit and this relates to the registering/unregistering discussion from elsewhere in this PR (specifically the one in https://github.com/PowerGridModel/power-grid-model/pull/1107/#discussion_r2316113388)

i've described the fact that the registering/unregistering is quite tricky in itself. an alternative could be to use `model.calculate(..., log_)`. Now we can also do a hybrid situation here: the adapter could be the boundary between registering/unregistering (to the job adapter). The model itself could be `model.calculate(..., log_)`. Then, the adapter still contains a logger, but the model no longer does.

thoughts?

Reply:
Experiment in https://github.com/PowerGridModel/power-grid-model/pull/1115